common/src/main/java/com/wynntils/core/statistics/StatisticsManager.java

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright © Wynntils 2023.
+ * This file is released under AGPLv3. See LICENSE for full license details.
+ */
+package com.wynntils.core.statistics;
+
+import com.wynntils.core.components.Manager;
+import com.wynntils.core.components.Models;
+import com.wynntils.core.storage.Storage;
+import com.wynntils.models.character.event.CharacterUpdateEvent;
+import com.wynntils.models.damage.type.DamageDealtEvent;
+import com.wynntils.models.spells.event.SpellEvent;
+import com.wynntils.models.stats.type.DamageType;
+import com.wynntils.models.worlds.event.WorldStateEvent;
+import java.util.EnumMap;
+import java.util.List;
+import java.util.Map;
+import java.util.TreeMap;
+import net.minecraftforge.eventbus.api.SubscribeEvent;
+
+// This should really be an "ExternalModel"...
+public final class StatisticsManager extends Manager {
+    // All statistics, per character
+    private final Storage<Map<String, Map<StatisticKind, Integer>>> statistics = new Storage<>(new TreeMap<>());
+
+    // The currently active statistics
+    private Map<StatisticKind, Integer> currentStatistics = new EnumMap<>(StatisticKind.class);
+
+    public StatisticsManager() {
+        super(List.of());
+    }
+
+    @SubscribeEvent
+    public void onWorldChange(WorldStateEvent event) {
+        // If we do not have a proper character, set up a fake statistics map so we always
+        // have a valid map.
+        if (!Models.Character.hasCharacter()) {
+            currentStatistics = new EnumMap<>(StatisticKind.class);
+            return;
+        }
```

Comment:
Is this just a crash safety measure?

Reply:
Yep, I tried to explain that in the comment but I can try to be more clear about it